client/app/scripts/charts/nodes-layout.js

```diff
@@ -487,7 +487,7 @@ export function doLayout(immNodes, immEdges, opts) {
 
     // cache results
     cache.cachedLayout = layout;
-    cache.nodeCache = cache.nodeCache.merge(layout.nodes);
+    cache.nodeCache = cache.nodeCache.merge(layout.nodes.map(n => fromJS({ x: n.get('x'), y: n.get('y') })));
```

Comment:
I was looking for `pick` in immutablejs but it doesn't exist. I'm fine either way but in this case here it would be

```js
cache.nodeCache = cache.nodeCache.merge(layout.nodes.map(n => fromJS(pick(n.toJS(), ['x', 'y']))));
```
which is less readable IMHO than your initial suggestion. I also don't see the robustness advantage since `get() `deals with missing properties as well.